### PR TITLE
Add quarterly business review demo

### PR DESCRIPTION
**Pull request by mcodik**

### CODE REVIEW

File: events/next18/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+* Extracts template variables out of a Google Slides template presentation
+* Writes those variables to a Google Sheets spreadsheet
+* Adds data to the spreadsheet based on those variables from a stub data service
+* Generates new Google Slides presentations using the template and the
+  spreadsheet data
+
+## Getting started
+
+* Follow the [Sheets API python quickstart](https://developers.google.com/sheets/api/quickstart/python)
+  * Make sure to save the `client-secrets.json` file in your working directory
```
[the remark this answers is not in the document]
Comment (mcodik) [reply events/next18/README.md]: done!

File: events/next18/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* Run the tool:
+
+<pre>
+    // Create the spreadsheet from the Google Slides template.
+    // For example, 13My9SxkotWssCc2F5yaXp2fzGrzoYV6maytr3qAT9GQ
+    $ python qbr_tool.py create_sheet --template_id &lt;your template id&gt;
+
+    // Add data from the stub customer service
+    $ python qbr_tool.py add_customers \
+        --spreadsheet_id &lt;your spreadsheet id&gt; \
```
[the remark this answers is not in the document]
Comment (mcodik) [reply events/next18/README.md]: done!

File: events/next18/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Getting started
+
+* Follow the [Sheets API python quickstart](https://developers.google.com/sheets/api/quickstart/python)
+  * Make sure to save the `client-secrets.json` file in your working directory
+* Enable the Google Slides API, Google Drive API and Google Sheets API in your
+  developer project
+* Run the tool with no arguments to complete the OAuth consent flow:
+
+<pre>
```
[the remark this answers is not in the document]
Comment (mcodik) [reply events/next18/README.md]: done!

File: events/next18/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    $ python qbr_tool.py
```
[the remark this answers is not in the document]
Comment (mcodik) [reply events/next18/README.md]: done!

File: events/next18/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+</pre>
+
+* Run the tool:
+
+<pre>
+    // Create the spreadsheet from the Google Slides template.
```
[the remark this answers is not in the document]
Comment (mcodik) [reply events/next18/README.md]: done!

File: events/next18/customer_spreadsheet_reader.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                'metadataKey': 'column_id',
+                'metadataValue': column_id,
+            }
+        }
+        self._data_filters[column_id] = data_filter
+
+    def ExecuteRead(self):
+        filters = list(self._data_filters.values())
+        get_body = {'dataFilters': filters}
+        read_fields = ('sheets.properties.sheetId,sheets.data.rowData.' +
```
[the remark this answers is not in the document]
Comment (mcodik) [reply events/next18/customer_spreadsheet_reader.py]: done!

File: events/next18/customer_spreadsheet_reader.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        return sheet.get('properties').get('sheetId')
+
+    def GetTemplateId(self):
+        metadata = self._spreadsheet.get('developerMetadata')[0]
+        return metadata.get('metadataValue')
+
+    def GetColumnData(self, column_id):
+        index = list(self._data_filters.keys()).index(column_id)
+        data = self._spreadsheet.get('sheets')[0].get('data')[index]
+        values = []
```
[the remark this answers is not in the document]
Comment (mcodik) [reply events/next18/customer_spreadsheet_reader.py]: done!

File: events/next18/presentation_writer.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+                'containsText': {
+                    'text': find_text,
+                    'matchCase': True
+                }
+            }
+        }
+        self._requests.append(request)
+
+    def ExecuteBatchUpdate(self):
+        body = {'requests': self._requests}
```
[the remark this answers is not in the document]
Comment (mcodik) [reply events/next18/presentation_writer.py]: I havent tested it in a while, but the intent is for this to be OK for ~thousands of placeholders.

**Comment by mcodik** [comment]
added a commit that addresses the feedback, thanks for taking a look!

File: events/next18/qbr_tool.py [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+import presentation_reader
+import presentation_writer
+import spreadsheet_writer
+
+
+SCOPES = ['https://www.googleapis.com/auth/drive']
+store = file.Storage('credentials.json')
+creds = store.get()
+if not creds or creds.invalid:
+    flow = client.flow_from_clientsecrets('client_secret.json', SCOPES)
```
Comment (erickoledadevrel) [inline events/next18/qbr_tool.py]: Need to update this to `credentials.json` as well.

**Comment by grant** [comment]
Note: I've fixed the Travis builds since the creation of this PR.
Please be sure the build passes before merging.

**Comment by mcodik** [comment]
Travis is now green, ptal!

**Pull request merged**